server.js

```diff
@@ -14,10 +21,81 @@ const welcomeMessage = {
 //This array is our "data store".
 //We will start with one message in the array.
 //Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
+let messages = [welcomeMessage];
 
-app.get("/", function (request, response) {
+app.get("/", function (request, response)
+{
   response.sendFile(__dirname + "/index.html");
 });
 
 app.listen(process.env.PORT);
+
+
+
+
+
+app.get("/messages", (req, res) =>
+{
+  res.send(messages);
+});
+
+app.post("/messages", (req, res) =>
+{
+  let msgName = req.body.from;
+  let msgText = req.body.text;
+  let idPosition = messages.length;
+
+  const newMsg =
+  {
+    id: idPosition,
+    name: msgName,
+    message: msgText,
+    timeSent: new Date(),
+  };
+
+  if (!newMsg.name || !newMsg.message)
+  {
+    return res.status(400).json("Please include a name and message");
+  }
+
+  else
```

Comment:
You don't need an else statement on this conditional, because the top level if statement contains a return statement. 

So you can do:

`if (!newMsg.name || !newMsg.message)
  {
    return res.status(400).json("Please include a name and message");
  }
 messages.push(newMsg);
 res.status(200).json(messages);
`
instead. Fewer lines = fewer bugs.